ProjectLibertyLabs/gateway
# Added spellcheck for markdown files

# Problem

tpyos

# Solution

Added `spellchecker-cli`, created its config and dictionary override files, and added to the github build workflow.

I only fixed clear errors.
Others may want to review the dictionary where there are multiple styles in use for the same term (e.g. NodeJS vs. Nodejs vs. nodejs).

Note: `docs/src/CoreConcepts/README.md` is excluded because it contains lorem ipsum text, and that seemed too silly to include in the dictionary.
Otherwise, all `.md` files should be found and checked.

## Steps to Verify:

- `npm run spellcheck` to run locally
- try adding a misspelling in one of the markdown files and see if it picks it up
- try submitting a PR with a misspelling and see if it breaks the build
